src/paradigma/heart_rate/heart_rate_analysis.py

```diff
@@ -85,4 +87,60 @@ def signal_quality_classification_io(input_path: Union[str, Path], output_path:
     metadata_time, metadata_values = read_metadata(input_path, config.meta_filename, config.time_filename, config.values_filename)
     df = tsdf.load_dataframe_from_binaries([metadata_time, metadata_values], tsdf.constants.ConcatenationType.columns)
 
-    df = signal_quality_classification(df, config, path_to_classifier_input)
\ No newline at end of file
+    df = signal_quality_classification(df, config, path_to_classifier_input)
+
+
+def estimate_heart_rate(df: pd.DataFrame, df_ppg_preprocessed: pd.DataFrame, config:HeartRateExtractionConfig) -> pd.DataFrame:  
```

Comment:
To me it's a little ambiguous as to the meaning of `df`. Could you perhaps choose a different naming for `df`, and expand on the difference between these two dataframes in the docstrings?